nats.go

```diff
@@ -3635,13 +3673,38 @@ func (s *Subscription) handleControlMessage(msg *Msg) (bool, error) {
 		if err != nil {
 			return isControl, err
 		}
+	} else {
+		// Process heartbeat received, get latest control metadata if present.
+		var ctrl *controlMetadata
+		cmeta := jsi.cmeta.Load()
+		if cmeta == nil {
+			return isControl, nil
+		}
+
+		ctrl = cmeta.(*controlMetadata)
+		tokens, err := getMetadataFields(ctrl.meta)
+		if err != nil {
+			return isControl, err
+		}
+		// Consumer sequence
+		dseq := tokens[6]
+		ldseq := msg.Header.Get(lastConsumerSeqHdr)
+
+		// Detect consumer sequence mismatch and whether
+		// should restart the consumer.
+		if ldseq != dseq {
+			// Dispatch async error including the sequence
+			// from where the consumer could be restarted.
+			sseq := parseNum(tokens[5])
+			handleConsumerSequenceMismatch(s, uint64(sseq+1))
```

Comment:
If not consumer sequence maybe the gap size or something?

Reply:
Added the consumer sequence and gap size as part of the error.